src/scripts/main.js

```diff
@@ -1,3 +1,258 @@
 'use strict';
 
-// write your code here
+const boardSize = 4;
+
+const boardBody = document.querySelector('.game-field').firstElementChild;
+const score = document.querySelector('.game-score');
+const gameButton = document.querySelector('.start');
+const cells = document.querySelectorAll('td');
+
+let board = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+for (let r = 0; r < boardSize; r++) {
+  for (let c = 0; c < boardSize; c++) {
+    const cell = boardBody.children[r].children[c];
+    const cellNumber = board[r][c];
+
+    cell.id = r.toString() + '-' + c.toString();
+
+    updateCell(cell, cellNumber);
+  }
+}
+
+function updateCell(currentCell, number) {
+  currentCell.innerText = '';
+  currentCell.classList.value = 'field-cell';
+
+  if (number > 0) {
+    currentCell.innerText = number;
+    currentCell.classList.add(`field-cell--${number}`);
+  }
+}
+
+gameButton.addEventListener('click', gameClick);
+
+function gameClick(e) {
+  const messages = document.querySelectorAll('.message');
+
+  messages.forEach(msg => msg.classList.add('hidden'));
+
+  if (gameButton.innerText === 'Start') {
+    window.addEventListener('keydown', start);
+    addNewNumber();
+    addNewNumber();
+
+    gameButton.removeEventListener('click', gameClick);
+  } else {
+    window.removeEventListener('keydown', start);
+
+    cells.forEach(cell => {
+      cell.innerText = '';
+      cell.classList.value = 'field-cell';
+    });
+
+    board = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
```

Comment:
you already have board, try to avoid duplicating, i think it can be done in other way